### PR TITLE
Frequency lower bound and target_neff shift in mode solver

### PR DESCRIPTION
**Pull request by weiliangjin2021**

### CODE REVIEW

File: tidy3d/plugins/mode/solver.py [hunk 1]
```diff
@@ -168,6 +170,12 @@ def compute_modes(
             target = mode_spec.target_neff
         target_neff_p = target / np.linalg.norm(kp_to_k) + fp_eps
```
Comment (momchil-flex) [inline tidy3d/plugins/mode/solver.py:171]: let's remove this `fp_eps` shift here then; it maybe was meant to play the same purpose but doesn't always?

File: tidy3d/plugins/mode/solver.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 TOL_EIGS = fp_eps
 # Tolerance for deciding on the matrix to be diagonal or tensorial
 TOL_TENSORIAL = 1e-6
+# shift target neff by this value, both rel and abs, whichever results in larger shift
+TARGET_SHIFT = -10 * fp_eps
```
Comment (momchil-flex) [inline tidy3d/plugins/mode/solver.py:22]: why negative?

File: tidy3d/plugins/mode/solver.py (hunk 1 above)
Comment (weiliangjin2021) [reply tidy3d/plugins/mode/solver.py:171]: it might be too small and overflowed.

File: tidy3d/plugins/mode/solver.py (hunk 2 above)
Comment (weiliangjin2021) [reply tidy3d/plugins/mode/solver.py:22]: i think either positive or negative is fine. Just randomly picked one.

File: tidy3d/plugins/mode/solver.py (hunk 1 above)
Comment (momchil-flex) [reply tidy3d/plugins/mode/solver.py:171]: right, so let's remove it in favor of the new code.

**Comment by momchil-flex** [comment]
Also could you please add fixes to changelog.

File: tidy3d/plugins/mode/solver.py (hunk 2 above)
Comment (weiliangjin2021) [reply tidy3d/plugins/mode/solver.py:22]: Positive can be safer to have a positive n_eff?

Comment (momchil-flex) [reply tidy3d/plugins/mode/solver.py:22]: I don't really have any intuition about it

**Comment by momchil-flex** [comment]
Looks good, please squash and merge.

**Pull request merged**